src/pages/api/delegate/mint-id.ts

```diff
@@ -28,13 +29,72 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
     if (!walletClient) {
       return;
     }
-    const transaction = await walletClient.writeContract({
-      address: config.contracts.talentLayerId,
-      abi: TalentLayerID.abi,
-      functionName: 'mintForAddress',
-      args: [userAddress, process.env.NEXT_PUBLIC_PLATFORM_ID, handle],
-      value: BigInt(handlePrice),
-    });
+
+    let transaction;
+
+    /**
+     * If addDelegateAndTransferId is true, we mint the TlId for the user, add the delegator address as delegate, then transfer it to them.
+     * @dev: This requires the delegator address not to have an existing TlId.
+     */
+    if (addDelegateAndTransferId) {
```

Comment:
it misses that NEXT_PUBLIC_ACTIVATE_DELEGATE_ON_MINT is active too, else an attacker can query it with true manually

Reply:
The check is done earlier L22